Tests/LibC/TestCComplex.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2023, Peter Elliott <pelliott@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <complex.h>
+#include <math.h>
+#include <stdio.h>
+
+// This file has to be C. C++ compilers will not compile c99 complex numbers.
+
+#if defined(AK_COMPILER_CLANG)
+#    pragma clang optimize off
+#else
+#    pragma GCC optimize("O0")
+#endif
+
+#define EXPECT_EQ(a, b)                                                                 \
+    {                                                                                   \
+        typeof(a) lhs = a;                                                              \
+        typeof(b) rhs = b;                                                              \
+        if (lhs != rhs) {                                                               \
+            fprintf(stderr, "\033[31;1mFAIL\033[0m: %s:%i: EXPECT_EQ(%s, %s) failed\n", \
+                __FILE__, __LINE__, #a, #b);                                            \
+            fail_counter++;                                                             \
+        }                                                                               \
+    }
+
+#define EXPECT_APPROXIMATE_WITH_ERROR(a, b, err)                                                 \
+    {                                                                                            \
+        typeof(a) lhs = a;                                                                       \
+        typeof(b) rhs = b;                                                                       \
+        if (cabsl(lhs - rhs) > err) {                                                            \
```

Comment:
This assumes that `cabsl` already works while testing `cabsl`. A silly implementation like `double cabsl(complex foo) { return -1.0; }` would probably pass this test.

Can this be avoided? What do you think about something like this:
`if (absl(creall(a) - creall(b)) > err || absl(cimagl(a) - cimagl(b)) > err) {`

Reply:
done